docs/docs/code/python/README.md

```diff
@@ -0,0 +1,279 @@
+
+For example, when your trigger step accepts data from an incoming HTTP request you may need to send that data to another API, transform that data or store it somewhere.
+
+In this example, we'll pretend this data is coming into our HTTP trigger.
+
+```json
+{
+  "id": 1,
+  "name": "Bulbasaur",
+  "type": "plant"
+}
+```
+
+Every Python step can import the `pipedream.script_helpers` module. This module contains data from other steps in the workflow via `steps` dictionary.
+
+Here's an example of how to get the ID from the Pokemon we received in the trigger.
+
+```python
+from pipedream.script_helpers import (steps, export)
+
+# retrieve the data points from the HTTP request in the initial workflow trigger 
+name = steps["trigger"]["event"]["name"]
+pokemon_type = steps["trigger"]["event"]["type"]
+
+print(f"{pokemon_name} is a {pokemon_type} type Pokemon")
+```
+
+### Sending data downstream to other steps
+
+The trigger steps automatically export their data for other steps to use downstream.
+
+Just pass the data to the `export` module from `pipedream.script_helpers`.
+
+```python
+# this step is named "code" in the workflow
+from pipedream.script_helpers import (steps, export)
+
+pokemon = { name: "Pikachu" }
+
+export('pokemon', pokemon)
+```
+
+Now this `pokemon` is accessible to downstream steps within `steps.code.pokemon`
+
+::: warning
+Not all data types can be stored in the `steps` data shared between workflow steps.  Only JSON serializable types can be passed, which includes:
+
+* lists 
+* dictionaries
+  :::
+
+## Using environment variables
+
+You can leverage any [environment variables defined in your Pipedream account](/environment-variables/#environment-variables) in a python step. This is useful for keeping your secrets out of code as well as keeping them flexible to swap API keys without having to update each step individually.
+
+To access them, use the `os` module.
+
+```python
+import os
+import requests
+
+token = os.environ['TWITTER_API_KEY']
+
+url = 'https://api.twitter.com/2/users/@pipedream/mentions'
+
+headers { 'Authorization': f"Bearer {token}"}
+r = requests.get(url, headers=headers)
+
+print(r.text)
+```
+
+:::tip
+There are 2 different ways of using the `os` module to access your environment variables.
+
+`os.environ['ENV_NAME_HERE']` will raise an error that stops your workflow if that key doesn't exist in your Pipedream account.
+
+Whereas `os.environ.get('ENV_NAME_HERE')` will _not_ throw an error and instead returns an empty string. 
+
+If your code relies on the presence of a environment variable, consider using `os.environ['ENV_NAME_HERE']` instead.
+:::
+
+## Handling errors
+
+You may find a need to exit a workflow early. In a Python step, just a `raise` an error to halt a step's execution.
+
+
+```python
+raise NameError('Something happened that should not. Exiting early.')
+```
+
+All exceptions from your Python code will appear in the logs area of the results.
```

Comment:
We should settle on a single way to discuss the **Results** and **Logs** section. Since Google's style guide recommends bolding UI elements, I would recommend bolding and capitalizing in all cases.

Reply:
True, perhaps a fast follow could be we link all mentions of Logs & Results in one section.

Side note, Google's style guide? Is that part of the build or some other tool?